api/v1alpha1/preflightvalidation_types.go

```diff
@@ -0,0 +1,116 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	VerificationTrue    string = "True"
+	VerificationFalse   string = "False"
+	VerificationStageImage   string = "Image"
+	VerificationStageBuild string = "Build"
+)
+
+// PreflightValidationSpec describes the desired state of the resource, such as the OCP image
+// that SR CRs need to be verified against and the debug configuration of the logs
+// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+// +kubebuilder:validation:Required
+type PreflightValidationSpec struct {
+	// Debug enables additional logging.
+	// +kubebuilder:validation:Optional
+	Debug bool `json:"debug"`
```

Comment:
I understand we will be exposing the validation status for each `Module`. Do we really need that boolean then? What will it govern? I think might be confusing to have it trigger more logs on the standard output - log level should be configurable in one place only.

Reply:
The idea was to enable more logs for this specific Preflight run. Since each Preflight run is eventually final, if there is any issues that we want to debug with logs, we can just apply a new Preflight CR with debug enabled

Reply:
> each Preflight run is eventually final

I think it should rather run continuously, each time a `Module` changes. WDYT?

Reply:
I am not sure about it ( from user's POV). Preflight CR should be applied once a week at best, probably even less frequently. I think once it has finished, user will simply delete, and create a new one next time, instead of just keeping it, and remembering that it is there and then updating it. Also, what happens if a Module CR changes, but the preflight has already been fully reconciled. Do we start Preflight reconciliation again? But user will not see the result, since he is not monitoring the Preflight

Reply:
fixed